### ISSUE TITLE
Rule proposal: JavaScript Expression spacing

### ISSUE THREAD

**Issue by neciu** [yannickcr/eslint-plugin-react#1007]
This rule should force adding/removing of spaces in JavaScript Expressions. 

Given the value of the rule is `always`:
1. Is wrong
```
...
  return (
    <div>
      {children}
    </div>
  );
...
```
2. Is ok
```
...
  return (
    <div>
      { children }
    </div>
  );
...
```

Given the value of the rule is `never`:
1. Is ok
```
...
  return (
    <div>
      {children}
    </div>
  );
...
```
2. Is wrong
```
...
  return (
    <div>
      { children }
    </div>
  );
...
```

**Comment by ljharb** [comment]
This is a duplicate of https://github.com/yannickcr/eslint-plugin-react/issues/857#issuecomment-255827739

**Comment by fatfisz** [comment]
This is fixed in #1177.
